File: app/src/main/java/com/akrivonos/a2chparser/pojomodel/threadmodel/Thread.java
package com.akrivonos.a2chparser.pojomodel.threadmodel;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Thread {

    @SerializedName("comment")
    @Expose
    private String comment;
    @SerializedName("lasthit")
    @Expose
    private Integer lasthit;
    @SerializedName("num")
    @Expose
    private String num;
    @SerializedName("posts_count")
    @Expose
    private Integer postsCount;
    @SerializedName("score")
    @Expose
    private Double score;
    @SerializedName("subject")
    @Expose
    private String subject;
    @SerializedName("timestamp")
    @Expose
    private Integer timestamp;
    @SerializedName("views")
    @Expose
    private Integer views;

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    public Integer getLasthit() {
        return lasthit;
    }

    public void setLasthit(Integer lasthit) {
        this.lasthit = lasthit;
    }

    public String getNum() {
        return num;
    }

    public void setNum(String num) {
        this.num = num;
    }

    public Integer getPostsCount() {
        return postsCount;
    }

    public void setPostsCount(Integer postsCount) {
        this.postsCount = postsCount;
    }

    public Double getScore() {
        return score;
    }

    public void setScore(Double score) {
        this.score = score;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public Integer getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(Integer timestamp) {
        this.timestamp = timestamp;
    }

    public Integer getViews() {
        return views;
    }

    public void setViews(Integer views) {
        this.views = views;
    }

}

File: app/src/main/java/com/akrivonos/a2chparser/pojomodel/threadmodel/Thread.kt
package com.akrivonos.a2chparser.pojomodel.threadmodel

import com.akrivonos.a2chparser.interfaces.FilteredItem
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Thread : FilteredItem{
    override fun getText(): String? = comment

    @SerializedName("banned")
    @Expose
    var banned: Int? = null
    @SerializedName("closed")
    @Expose
    var closed: Int? = null
    @SerializedName("comment")
    @Expose
    var comment: String? = null
    @SerializedName("date")
    @Expose
    var date: String? = null
    @SerializedName("email")
    @Expose
    var email: String? = null
    @SerializedName("endless")
    @Expose
    var endless: Int? = null
    @SerializedName("files")
    @Expose
    var files: List<File>? = null
    @SerializedName("files_count")
    @Expose
    var filesCount: Int? = null
    @SerializedName("lasthit")
    @Expose
    var lasthit: Int? = null
    @SerializedName("name")
    @Expose
    var name: String? = null
    @SerializedName("num")
    @Expose
    var num: String? = null
    @SerializedName("op")
    @Expose
    var op: Int? = null
    @SerializedName("parent")
    @Expose
    var parent: String? = null
    @SerializedName("posts_count")
    @Expose
    var postsCount: Int? = null
    @SerializedName("sticky")
    @Expose
    var sticky: Int? = null
    @SerializedName("subject")
    @Expose
    var subject: String? = null
    @SerializedName("tags")
    @Expose
    var tags: String? = null
    @SerializedName("timestamp")
    @Expose
    var timestamp: Int? = null
    @SerializedName("trip")
    @Expose
    var trip: String? = null

}
